webapp/src/routes/auth/signup/+page.svelte

```diff
@@ -0,0 +1,118 @@
+<script lang="ts">
+    import FormHeader from '$lib/components/form-header.svelte';
+    import SubmitButton from '$lib/components/submit-button.svelte';
+    import { BACKEND_SERVER, PASSWORD_NOT_MATCH } from '$lib/constants';
+    import { Form, TextInput, PasswordInput } from 'carbon-components-svelte';
+    let firstName: string;
+    let lastName: string;
+    let email: string;
+
+    let password: string;
+    let passwordConfirm: string;
+
+    let errorMessage: string;
+    let successMessage: string;
+    let accountCreated: boolean = false;
+    let valid: boolean = false;
+
+    $: {
+        if (password !== passwordConfirm) {
+            errorMessage = PASSWORD_NOT_MATCH;
+            valid = false;
+        } else {
+            errorMessage = '';
+            valid = true;
+        }
+    }
+
+    async function signup() {
+        if (valid) {
+            try {
+                const response = await fetch(`${BACKEND_SERVER}/user`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json'
+                    },
+                    body: JSON.stringify({
+                        full_name: `${firstName} ${lastName}`,
+                        email: email,
+                        password: password
+                    })
+                });
+
+                if (response.ok) {
+                    successMessage =
+                        'Account created successfully. Normally we would ask for email verification but for demo... nah';
+                    accountCreated = true;
+                    errorMessage = '';
+                } else {
+                    const data = await response.json();
+                    errorMessage = data.errors[0].message;
+                }
+            } catch {
+                errorMessage = 'Something went wrong in the server T_T';
```

Comment:
We would be changing these error messages later right?

Reply:
Yes lol